src/services/codefixes/interfaceFixes.ts

```diff
@@ -0,0 +1,268 @@
+                        textChanges: textChanges
+                    }]
+                }];
+            }
+
+            return undefined;
+        }
+    });
+
+    registerCodeFix({
+        errorCodes: [Diagnostics.Class_0_incorrectly_implements_interface_1.code],
+        getCodeActions: (context: CodeFixContext) => {
+            const sourceFile = context.sourceFile;
+            const start = context.span.start;
+            const token = getTokenAtPosition(sourceFile, start);
+            const checker = context.program.getTypeChecker();
+
+            let textChanges: TextChange[] = [];
+
+            if (token.kind === SyntaxKind.Identifier && token.parent.kind === SyntaxKind.ClassDeclaration) {
+                const classDeclaration = <ClassDeclaration>token.parent;
+                const startPos: number = classDeclaration.members.pos;
+                const classMembers = getClassMembers(classDeclaration);
+                const trackingAddedMembers: string[] = [];
+                const interfaceClauses = ts.getClassImplementsHeritageClauseElements(classDeclaration);
+
+                for (let i = 0; interfaceClauses && i < interfaceClauses.length; i++) {
+                    textChanges = textChanges.concat(getChanges(interfaceClauses[i], classMembers, startPos, checker, /*reference*/ false, trackingAddedMembers, context.newLineCharacter));
+                }
+            }
+
+            if (textChanges.length > 0) {
+                return [{
+                    description: getLocaleSpecificMessage(Diagnostics.Implement_interface_on_class),
+                    changes: [{
+                        fileName: sourceFile.fileName,
+                        textChanges: textChanges
+                    }]
+                }];
+            }
+
+            return undefined;
+        }
+    });
+
+    registerCodeFix({
+        errorCodes: [Diagnostics.Non_abstract_class_0_does_not_implement_inherited_abstract_member_1_from_class_2.code],
+        getCodeActions: (context: CodeFixContext) => {
+            const sourceFile = context.sourceFile;
+            const start = context.span.start;
+            const token = getTokenAtPosition(sourceFile, start);
+            const checker = context.program.getTypeChecker();
+
+            let textChanges: TextChange[] = [];
+
+            if (token.kind === SyntaxKind.Identifier && token.parent.kind === SyntaxKind.ClassDeclaration) {
+                const classDeclaration = <ClassDeclaration>token.parent;
+                const startPos = classDeclaration.members.pos;
+                const classMembers = getClassMembers(classDeclaration);
+                const trackingAddedMembers: string[] = [];
+                const extendsClause = ts.getClassExtendsHeritageClauseElement(classDeclaration);
+                textChanges = textChanges.concat(getChanges(extendsClause, classMembers, startPos, checker, /*reference*/ false, trackingAddedMembers, context.newLineCharacter));

```

Comment:
This does not handle properties declared in the constructor as parameter property declarations, e.g. this should only add `b` and not `a`.

````ts
interface I {
    a: number,
    b:string;
}
class C implements I {
    constructor(public a) {}
}
```

Please add tests.